azuredevops/internal/acceptancetests/resource_serviceendpoint_azuredevops_test.go

```diff
@@ -0,0 +1,179 @@
+					resource.TestCheckResourceAttr(tfSvcEpNode, "service_endpoint_name", serviceEndpointNameFirst),
+				),
+			},
+			{
+				Config: hclSvcEndpointAzureDevOpsResourceUpdate(projectName, serviceEndpointNameSecond, orgUrl, releaseApiUrl, description),
+				Check: resource.ComposeTestCheckFunc(
+					testutils.CheckServiceEndpointExistsWithName(tfSvcEpNode, serviceEndpointNameSecond),
+					resource.TestCheckResourceAttrSet(tfSvcEpNode, "project_id"),
+					resource.TestCheckResourceAttr(tfSvcEpNode, "org_url", orgUrl),
+					resource.TestCheckResourceAttr(tfSvcEpNode, "release_api_url", releaseApiUrl),
+					resource.TestCheckResourceAttr(tfSvcEpNode, "service_endpoint_name", serviceEndpointNameSecond),
+					resource.TestCheckResourceAttr(tfSvcEpNode, "description", description),
+				),
+			},
+		},
+	})
+}
+
+func TestAccServiceEndpointAzureDevOps_RequiresImportErrorStep(t *testing.T) {
+	projectName := testutils.GenerateResourceName()
+	serviceEndpointName := testutils.GenerateResourceName()
+	resourceType := "azuredevops_serviceendpoint_azuredevops"
+	tfSvcEpNode := resourceType + ".test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
+		Providers:    testutils.GetProviders(),
+		CheckDestroy: testutils.CheckServiceEndpointDestroyed(resourceType),
+		Steps: []resource.TestStep{
+			{
+				Config: hclSvcEndpointAzureDevOpsResourceBasic(projectName, serviceEndpointName),
+				Check: resource.ComposeTestCheckFunc(
+					testutils.CheckServiceEndpointExistsWithName(tfSvcEpNode, serviceEndpointName),
+				),
+			},
+			{
+				Config:      hclSvcEndpointAzureDevOpsResourceRequiresImport(projectName, serviceEndpointName),
+				ExpectError: testutils.RequiresImportError(serviceEndpointName),
+			},
+		},
+	})
+}
+
+func hclSvcEndpointAzureDevOpsResourceBasic(projectName string, serviceEndpointName string) string {
+	serviceEndpointResource := fmt.Sprintf(`
+resource "azuredevops_serviceendpoint_azuredevops" "test" {
+	project_id             = azuredevops_project.project.id
+	service_endpoint_name  = "%s"
+	personal_access_token  = "0000000000000000000000000000000000000000000000000000"
+}`, serviceEndpointName)
+
+	projectResource := testutils.HclProjectResource(projectName)
+	return fmt.Sprintf("%s\n%s", projectResource, serviceEndpointResource)
+}
+
+func hclSvcEndpointAzureDevOpsResourceComplete(projectName string, serviceEndpointName string, description string) string {
+	serviceEndpointResource := fmt.Sprintf(`
+resource "azuredevops_serviceendpoint_azuredevops" "test" {
+	project_id             = azuredevops_project.project.id
+	service_endpoint_name  = "%s"
+	description            = "%s"
+	org_url			   	   = "https://dev.azure.com/myorganization"
+	release_api_url		   = "https://vsrm.dev.azure.com/myorganization"
+	personal_access_token  = "0000000000000000000000000000000000000000000000000000"
```

Comment:
Would you run `terrafmt fmt ...` for it, install: https://github.com/katbyte/terrafmt